check-services.php

```diff
@@ -61,10 +66,59 @@
     }
 }
 
-$sql = 'SELECT * FROM `devices` AS D, `services` AS S WHERE S.device_id = D.device_id ' . $where . ' ORDER by D.device_id DESC';
+$sql = 'SELECT * FROM `devices` AS D'
+       .' INNER JOIN `services` AS S ON S.device_id = D.device_id AND D.disabled = 0 '.$where
+       .' LEFT JOIN `devices_attribs` as A ON D.device_id = A.device_id AND A.attrib_type = "override_icmp_disable"'
+       .' ORDER by D.device_id DESC;';
 
 foreach (dbFetchRows($sql) as $service) {
-    // Run the polling function
-    poll_service($service);
-} //end foreach
+    // Run the polling function if the associated device is up, "Disable ICMP Test" option is not enabled,
+    // or service hostname/ip is different from associated device
+    if ($service['status'] === "1" || ($service['status'] === '0' && $service['status_reason'] === 'snmp') ||
+        $service['attrib_value'] === 'true' || ($service['service_ip'] !== $service['hostname'] &&
+        $service['service_ip'] !== inet6_ntop($service['ip']) )) {
+        // Mark service check as enabled if it was disabled previously because device was down
+        if ($service['service_disabled'] === "1") {
+            dbUpdate(
+                array('service_disabled' => '0'),
+                'services',
+                '`service_id` = ?',
+                array($service['service_id'])
+            );
+        }
+        poll_service($service);
+        $polled_services++;
+    } else {
+        d_echo("\nNagios Service - ".$service['service_id']."\nSkipping service check because device "
+               .$service['hostname']." is down due to icmp.\n");
+        // Mark service check as disabled while device is down and log to eventlog that service check is skipped,
+        // but only if it's not already marked as disabled
+        if ($service['service_disabled'] === "0") {
+            dbUpdate(
+                array('service_disabled' => '1'),
+                'services',
+                '`service_id` = ?',
+                array($service['service_id'])
+            );
+            log_event(
+                "Nagios Service - {$service['service_desc']} ({$service['service_id']}) - 
```

Comment:
AFAIK, they shouldn't be called "Nagios Service"
https://www.monitoring-plugins.org/